modules/RTC/BridgeChannel.js

```diff
@@ -49,16 +49,15 @@ export default class BridgeChannel {
         // If a RTCPeerConnection is given, listen for new RTCDataChannel
         // event.
         if (peerconnection) {
-            peerconnection.ondatachannel = event => {
-                // NOTE: We assume that the "onDataChannel" event just fires
-                // once.
-
-                const datachannel = event.channel;
-
-                // Handle the RTCDataChannel.
-                this._handleChannel(datachannel);
-                this._mode = 'datachannel';
-            };
+            const datachannel
+                = peerconnection.createDataChannel(
```

Comment:
This is the client-side part of this PR in the bridge https://github.com/jitsi/jitsi-videobridge/pull/723. The bridge PR has a nice explanation of why we're moving the logic in the client and maybe it'd be useful to include it as a comment in the code.

Reply:
I have tried to explain it here:
https://github.com/jitsi/jitsi-videobridge/pull/723

The thing is that Chrome tries to gracefully close the data channels and will not release it's SID (0 - used by the bridge) until it gets the close response. When conference is abruptly moved to another bridge it has no knowledge about that data channel and it will not reply. When it tries to open the default channel on SID=0 again Chrome will not accept it (but there's no error returned to the bridge). It's more reliable to make the client open the data channels, so that it can select available SID every time the channel needs to be opened.

A SID is SCTP stream id. Some kind of a channel on the SCTP layer.

I have tried to make the bridge open channel on a random SID, but it was sometimes causing SCTP error which I could not understand and there's a risk of collision anyway (even though very small).